src/url.rs

```diff
@@ -28,16 +30,23 @@ impl Scanner for UrlScanner {
 
 impl UrlScanner {
     // See "scheme" in RFC 3986
-    fn find_start(&self, s: &str) -> Option<usize> {
+    fn find_start(&self, s: &str) -> (Option<usize>, Option<char>) {
```

Comment:
I used a tuple type as a return value here.
Alternatively we could introduce a `Marker` struct:

```rust
struct Marker {
  first_char: usize,
  quote: Option<char>,
}
```

And then return `Option<Marker>` here instead.